src/constructor.jl

```diff
@@ -26,15 +26,28 @@ mutable struct Constructor
     constructed_objects::Dict{Node, Any}
     recursive_objects::Set{Node}
     yaml_constructors::Dict{Union{String, Nothing}, Function}
+    yaml_multi_constructors::Dict{Union{String, Nothing}, Function}
 
-    function Constructor(more_constructors::Dict{String,Function})
+    function Constructor(single_constructors = Dict(), multi_constructors = Dict())
         new(Dict{Node, Any}(), Set{Node}(),
-            merge(copy(default_yaml_constructors), more_constructors))
+            convert(Dict{Union{String, Nothing},Function}, single_constructors),
+            convert(Dict{Union{String, Nothing},Function}, multi_constructors))
     end
 end
 
-Constructor() = Constructor(Dict{String,Function}())
+
+function add_constructor!(func::Function, constructor::Constructor, tag::Union{String, Nothing})
```

Comment:
Could this and other instances of `::Function` be removed to allow for other callable stuff?

Reply:
It probably could, I was following what was already there.